backend/controllers/taskController.js

```diff
@@ -89,4 +90,50 @@ const getTasksByDay = asyncHandler(async (req, res) => {
   res.status(200).json(tasks);
 });
 
-module.exports = { createTask, getTasks, getTasksByDay, getTasksById };
+const getTasksByMonth = asyncHandler(async (req, res) => {
+  const userId = req.id;
+
+  // Access the provided 'start' and 'end' query parameters
+  let start = req.params.month;
+  let startDate = 0;
+  let endDate = 0;
+
+  if (start.length === 6) {
+    // parse
+    const year_s = parseInt(start.substring(0, 4));
+    const month_s = parseInt(start.substring(4, 6));
+
+    try {
+      startDate = new Date(year_s, month_s - 1, 1);
+      endDate = new Date(year_s, month_s, 1);
+    } catch (error) {
+      throw new Error("Invalid Date Input");
+    }
+  }
+
+  // Check if its a valid date object
+  const tasks = await Task.find({
+    user_id: userId,
+    startDate: {
+      $gte: startDate,
+      $lte: endDate,
+    },
+    isStarted: true,
```

Comment:
I think for isStarted, we can change this to where the task actually doesn't need to be started to fetch it. The same change should also be done to the getTasksByDay function